code/game/objects/random/subtypes/closets.dm

```diff
@@ -0,0 +1,28 @@
+/obj/random/closet
+	name = "random closet"
+	desc = "This is a random closet."
+	icon = 'icons/obj/closets/bases/closet.dmi'
+	icon_state = "base"
+	var/vermin_chance = 0.1
+	var/list/locker_vermin = list(
+		/mob/living/simple_animal/mouse,
+		/mob/living/simple_animal/opossum
+	)
+
+/obj/random/closet/spawn_choices()
+	return typesof(/obj/structure/closet) - (typesof(/obj/structure/closet/crate) | typesof(/obj/structure/closet/body_bag) | typesof(/obj/structure/closet/secure_closet))
```

Comment:
This could be a nontrivial amount of init time. Maybe make this a static list? Something like:
```suggestion
	var/static/list/spawn_types = typesof(/obj/structure/closet) - (typesof(/obj/structure/closet/crate) | typesof(/obj/structure/closet/body_bag) | typesof(/obj/structure/closet/secure_closet))
	return spawn_types
```